llvm/tools/sycl-post-link/sycl-post-link.cpp

```diff
@@ -295,20 +438,73 @@ saveResultModules(std::vector<std::unique_ptr<Module>> &ResModules) {
   return Res;
 }
 
-static string_vector
-saveSpecConstantIDMaps(const std::vector<SpecIDMapTy> &Maps) {
-  string_vector Res;
+// Each fallback device library corresponds to one bit in "require mask" which
+// is an unsigned int32. getDeviceLibBit checks which fallback device library
+// is required for FuncName and returns the corresponding bit. The corresponding
+// mask for each fallback device library is:
+// fallback-cassert:      0x1
+// fallback-cmath:        0x2
+// fallback-cmath-fp64:   0x4
+// fallback-complex:      0x8
+// fallback-complex-fp64: 0x10
+static uint32_t getDeviceLibBits(const std::string &FuncName) {
+
+  if (DeviceLibFuncMap.count(FuncName) == 0)
+    return 0;
+  else
+    return 0x1 << (DeviceLibFuncMap[FuncName] - cl_intel_devicelib_assert);
+}
+
+// For each device image module, we go through all functions which meets
+// 1. The function name has prefix "__devicelib_"
+// 2. The function has SPIR_FUNC calling convention
+// 3. The function is declaration which means it doesn't have function body
+static uint32_t getModuleReqMask(const Module &M) {
+  // 0x1 means sycl runtime will link and load libsycl-fallback-assert.spv as
```

Comment:
Suggest to check Module target triple being spirv instead of filtering by calling convention below.

Reply:
Hi, @kbobrovs 
I added check for target triple in this function, the logic is we will return "0"(means no devicelib is needed) if target triple is NOT spirv. But I am not sure whether I used the correct way to check the target triple string, I assumed that for spirv target, the target triple string should have the prefix "spir64-", is it a valid assumption?
Thank you very much.

Reply:
I suggest `llvm::Triple(M.getTargetTriple()).isSPIR()`

Reply:
Done.